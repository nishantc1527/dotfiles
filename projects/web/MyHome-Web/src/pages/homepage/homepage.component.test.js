import React from "react";
import {render} from "@testing-library/react";
import HomePage from "./homepage.component";
import {Provider} from "react-redux";
import configureStore from 'redux-mock-store';

describe('Home Page', () => {

    const mockStore = configureStore();
    const initialData = {
        user: {
            currentUser: null
        }
    };
    const loggedInUserData = {
        user: {
            currentUser: {
                userId: 'any-uuid',
                token: 'any-token'
            }
        }
    };

    const storeWithNoUserData = mockStore(initialData);
    const storeWithUserData = mockStore(loggedInUserData);

    it('should match snapshot test for home page component', () => {
        expect(render(<Provider store={storeWithNoUserData}>
            <HomePage currentUser={initialData.user.currentUser}/>
        </Provider>)).toMatchSnapshot()
    });

    it('should render welcome message only if user is not having token', () => {
        const { queryByTestId } = render(<Provider store={storeWithNoUserData}>
            <HomePage currentUser={initialData.user.currentUser}/>
        </Provider>);

        expect(queryByTestId('welcome-message')).toBeTruthy();
    });

    it('should not render welcome message if user is having token', () => {
        const { queryByTestId } = render(<Provider store={storeWithUserData}>
            <HomePage currentUser={loggedInUserData.user.currentUser} />
        </Provider>)

        expect(queryByTestId('welcome-message')).toBeFalsy();
    });
});
