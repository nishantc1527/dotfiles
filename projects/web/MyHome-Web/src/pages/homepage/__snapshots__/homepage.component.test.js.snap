// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home Page should match snapshot test for home page component 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div>
        <div
          class="container text-center"
          data-testid="welcome-message"
        >
          <p
            class="h5 mt-5"
          >
            Welcome to MyHome Web Application
            <br />
            This Application helps people manage their apartments
            <br />
            <a
              href="signin"
            >
              Login
            </a>
             to continue
          </p>
        </div>
        
      </div>
    </div>
  </body>,
  "container": <div>
    <div>
      <div
        class="container text-center"
        data-testid="welcome-message"
      >
        <p
          class="h5 mt-5"
        >
          Welcome to MyHome Web Application
          <br />
          This Application helps people manage their apartments
          <br />
          <a
            href="signin"
          >
            Login
          </a>
           to continue
        </p>
      </div>
      
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
